rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -215,83 +215,43 @@ ROCAL_API_CALL rocalGetImageId(RocalContext p_context,  int* buf)
     }
 }
 
-void
-ROCAL_API_CALL rocalGetImageLabels(RocalContext p_context, void* buf, unsigned int flags)
+RocalTensorList
+ROCAL_API_CALL rocalGetImageLabels(RocalContext p_context)
 {
 
     if (!p_context)
         THROW("Invalid rocal context passed to rocalGetImageLabels")
     auto context = static_cast<Context*>(p_context);
-    auto meta_data = context->master_graph->meta_data();
-    if(!meta_data.second) {
-        WRN("No label has been loaded for this output image")
-        return;
-    }
-    size_t meta_data_batch_size = meta_data.second->get_label_batch().size();
-    if(context->user_batch_size() != meta_data_batch_size)
-        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
-
-    if (context->affinity == RocalAffinity::CPU)
-        memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
-    else
-    {
-#if ENABLE_HIP
-        if (!flags) {
-            hipError_t err = hipMemcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size, hipMemcpyHostToDevice);
-            if (err != hipSuccess)
-                THROW("Invalid Data Pointer: Error copying to device memory")
-        }
-        else if (flags & ROCAL_MEMCPY_TO_HOST) {
-            memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
-        }
-#elif ENABLE_OPENCL
-        if (!flags) {
-          if(clEnqueueWriteBuffer(context->master_graph->get_ocl_cmd_q(), (cl_mem)buf, CL_TRUE, 0, sizeof(int) * meta_data_batch_size, meta_data.second->get_label_batch().data(), 0, NULL, NULL) != CL_SUCCESS)
-              THROW("Invalid Data Pointer: Error copying to device memory")
-        }
-        else if (flags & ROCAL_MEMCPY_TO_HOST) {
-            memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
-        }
-#endif
-    }
+    return context->master_graph->labels_meta_data();
```

Comment:
Do we not need to do error handling?

Reply:
Need to check once performance PR is merged on whether we need to retain flags and use them